### ISSUE TITLE
How to know TabsController tab is changed?

### ISSUE THREAD

**Issue by OrkhanAlikhanov** [CosmicMind/Material#859]
I was implementing `TabBarDelegate` on `tabsController.tabBar` to get notified when tab is changed. So you have filled that hole:
https://github.com/CosmicMind/Material/blob/63301f01f2b30519a4ba22b70dddb0ebcb8166a8/Sources/iOS/TabsController.swift#L262

I'd like to get controller-wide notification. I tried to override `selectedIndex` and implement `didSet`
```swift
override var selectedIndex: Int {
    didSet {
        print("I am set")
    }
}
```
I get error since it is marked as `private (set)`
>Cannot observe read-only property `selectedIndex`; it can't change.

So, any other way?

**Comment by danieldahan** [comment]
I can make a TabsControllerDelegate for you now. There should be one and this would be beneficial for everyone. Can you suggest what other items you may like in that delegate before I make it?

**Comment by OrkhanAlikhanov** [comment]
Basic things from [`UITabBarControllerDelegate`](https://developer.apple.com/documentation/uikit/uitabbarcontrollerdelegate) would be enough I guess.

**Comment by danieldahan** [comment]
Cool, I will add some basic insight. Thanks, this will be done within the next couple hours.

**Comment by OrkhanAlikhanov** [comment]
Thank you for your support 👍

**Comment by danieldahan** [comment]
In [Material 2.10.1](https://github.com/CosmicMind/Material/releases/tag/2.10.1) you can now use the following delegation methods: 

```swift
@objc(TabsControllerDelegate)
public protocol TabsControllerDelegate {
    /**
     A delegation method that is executed to determine if the TabsController should
     transition to the next view controller.
     - Parameter tabBar: A TabsController.
     - Parameter tabItem: A TabItem.
     - Returns: A Boolean.
     */
    @objc
    optional func tabsController(tabsController: TabsController, shouldSelect viewController: UIViewController) -> Bool
    
    /**
     A delegation method that is executed when the view controller will transitioned to.
     - Parameter tabsController: A TabsController.
     - Parameter viewController: A UIViewController.
     */
    @objc
    optional func tabsController(tabsController: TabsController, willSelect viewController: UIViewController)
    
    /**
     A delegation method that is executed when the view controller has been transitioned to.
     - Parameter tabsController: A TabsController.
     - Parameter viewController: A UIViewController.
     */
    @objc
    optional func tabsController(tabsController: TabsController, didSelect viewController: UIViewController)
}
```

Thank you!

**Comment by OrkhanAlikhanov** [comment]
Great ^^

For your information, you can make issues get closed automatically in commits. For example, you can commit like this: `Added TabsControllerDelegate. Closes #859`
The issue #859 will be automatically closed as soon as the commit becomes the part of master branch. [An example commit](https://github.com/showdownjs/showdown/commit/51e46933a483fb9324ac83c0751f5162833d2ebb)

[Here is docs](https://help.github.com/articles/closing-issues-using-keywords).

**Comment by danieldahan** [comment]
Nice I shall try that out :) Thank you!
